### DISCUSSION TITLE
Как в zapret повторить работу https://github.com/Flowseal/zapret-discord-youtube ?

### DISCUSSION THREAD

**Discussion by iqubik** [bol-van/zapret#852] [Q&A]
Не получается никак настроить zapret так же как настроен github.com/Flowseal/zapret-discord-youtube
Помогите пожалуйста

**Comment by zefkerr** [comment]
> Не получается никак настроить zapret так же как настроен github.com/Flowseal/zapret-discord-youtube Помогите пожалуйста

@iqubik  Не надо так делать. Не существует универсальных настроек, работающих одинаково у всех. В разных городах, для разных провайдеров необходимо использовать разные настройки. Кроме того, например, настроки, работающие в GNU/Linux зачастую нельзя использовать в Windows. Вам необходимо запустить скрипт ./blockcheck.sh, чтобы выяснить, какие настройки подходят индивидуально для вас. Попроще и поудобнее "копировать и вставить" настройки "от Васяна" здесь делать нет смысла. В указанном вами репозитории используются настройки в стиле "трэш-спам". То есть автор просто запихнул в конфиг всё подряд, ибо, вероятно, полагает, что чем больше, тем лучше будет работать. Но это не так. Даже один лишний параметр в конфиге может поломать всю работу полностью.

https://github.com/Flowseal/zapret-discord-youtube/blob/main/discord.bat

**Comment by iqubik** [comment]
Я попробую уже который раз подобрать к Билайн через скрипт, но прошлые 3 раза такой подбор не увенчался успехом :(
Потому я и хотел конфиг который точно работает перенести на роутер

**Comment by reidzi** [comment]
Максимально копипастно, но думать было лень.
Запихай файл из zapret-discord-youtube list-general.txt в /opt/zapret/ipset/list-general.txt
в файле /opt/zapret/config поменяй строчки:
NFQWS_OPT_DESYNC_HTTP="--hostlist="/opt/zapret/ipset/list-general.txt" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new"
NFQWS_OPT_DESYNC_HTTPS="--hostlist="/opt/zapret/ipset/list-general.txt" --dpi-desync=fake,split --dpi-desync-autottl=2 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls="/opt/zapret/files/fake/quic_initial_www_google_com.bin" --new"
NFQWS_OPT_DESYNC_QUIC="--hostlist="/opt/zapret/ipset/list-general.txt" --dpi-desync=fake,split --dpi-desync-autottl=2 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls="/opt/zapret/files/fake/quic_initial_www_google_com.bin" --new"
при установке install_easy.sh выбери select MODE : 3 : nfqws

с этим будет работать только ютуб. для дискорда нужны udp порты, а это надо доки читать. а я тупенький.

**Comment by iqubik** [comment]
> Максимально копипастно, но думать было лень.

Нет к сожалению на дискорд ни ютуб с таким конфигом у меня не взлетел. Там на базе zapret но видимо автор что-то сделал кастомное что и ютуб и дикорд летают. Может можно как то объединить идеи?

**Comment by stdafxh** [comment]
Билайн (Ростов-на-Дону):

nfqws --dpi-desync=split2 --dpi-desync-split-seqovl=1 --dpi-desync-split-tls=sni --wssize 1:6

**Comment by Elllkere** [comment]
> Там на базе zapret но видимо автор что-то сделал кастомное что и ютуб и дикорд летают

там и есть обычный запрет (по крайней мере информации о изменениях нет), просто батники собрали с готовыми пресетами. Если там работает, зачем пытаться в обычном запрете сделать тоже самое?

По такой же аналогии могу дать https://github.com/Elllkere/misterfishdpi (Windows) , интерфейс обертка для оригинального запрета

**Comment by aldari** [comment]
Вопрос стоит так: 
- zapret установлен и раньше работал в linux
- сейчас настройки изменились и хотелось бы обновить стратегию
- на windows на моем провайдере работает стратегия из general (ALT).bat 

как преобразовать настройки из bat в linux ?

--filter-udp=443 --hostlist="list-general.txt" --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new
--filter-udp=50000-50100 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-cutoff=d3 --dpi-desync-repeats=6 --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new
--filter-tcp=80 --hostlist="list-general.txt" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new
--filter-tcp=443 --hostlist="list-general.txt" --dpi-desync=fake,split --dpi-desync-autottl=5 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls="%BIN%tls_clienthello_www_google_com.bin


^ - символ перевода строки игнорируем
--new -  я игнорирую
" - из путей убрал

NFQWS_OPT_DESYNC_HTTP="--hostlist=/opt/zapret_lists/list-youtube.txt --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig"
NFQWS_OPT_DESYNC_HTTPS="--hostlist=/opt/zapret_lists/list-youtube.txt --dpi-desync=fake,split --dpi-desync-autottl=5 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls="/opt/zapret/files/fake/tls_clienthello_www_google_com.bin""
NFQWS_OPT_DESYNC_QUIC="--hostlist=/opt/zapret_lists/list-youtube.txt --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic="/opt/zapret/files/fake/quic_initial_www_google_com.bin""

Путь к bin указывал с кавычками, без них аргумент пропадает.

**Comment by iqubik** [comment]
> Билайн (Ростов-на-Дону):
> 
> nfqws --dpi-desync=split2 --dpi-desync-split-seqovl=1 --dpi-desync-split-tls=sni --wssize 1:6

Спасибо! А это работает. Ютуб пошёл. А вот дискорд не желает.

**Comment by userosos** [comment]
Так, ну и кто смог переписать это
```
@echo off
chcp 65001 >nul
:: 65001 - UTF-8

cd /d "%~dp0"

set BIN=%~dp0bin\

start "zapret: general" /min "%BIN%winws.exe" --wf-raw="@rules.txt" ^
--filter-udp=443 --hostlist="list-general.txt" --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new ^
--filter-udp=50000-65535 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-cutoff=d3 --dpi-desync-repeats=6 --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new ^
--filter-tcp=80 --hostlist="list-general.txt" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new ^
--filter-tcp=443 --hostlist="list-general.txt" --dpi-desync=fake,split --dpi-desync-autottl=2 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls="%BIN%tls_clienthello_www_google_com.bin"

```
rules.txt

```
!impostor and !loopback and
 ((outbound and (((tcp.DstPort == 80) or (tcp.DstPort == 443)) or ((udp.DstPort == 443) or 
				(udp.DstPort >= 50000 and udp.DstPort <= 65535 and udp.PayloadLength == 74 and udp.Payload32[0] == 0x00010046
																	))) and 
 (((ip.DstAddr < 127.0.0.1 or ip.DstAddr > 127.255.255.255) and (ip.DstAddr < 10.0.0.0 or ip.DstAddr > 10.255.255.255) and (ip.DstAddr < 192.168.0.0 or ip.DstAddr > 192.168.255.255) and (ip.DstAddr < 172.16.0.0 or ip.DstAddr > 172.31.255.255) and (ip.DstAddr < 169.254.0.0 or ip.DstAddr > 169.254.255.255)) or ((ipv6.DstAddr > ::1) and (ipv6.DstAddr < 2001::0 or ipv6.DstAddr >= 2001:1::0) and (ipv6.DstAddr < fc00::0 or ipv6.DstAddr >= fe00::0) and (ipv6.DstAddr < fe80::0 or ipv6.DstAddr >= fec0::0) and (ipv6.DstAddr < ff00::0 or ipv6.DstAddr >= ffff::0))))
  or
  (inbound and tcp and (tcp.Ack and tcp.Syn or tcp.Rst or tcp.Fin) and ((tcp.SrcPort == 80) or (tcp.SrcPort == 443)) and (((ip.SrcAddr < 127.0.0.1 or ip.SrcAddr > 127.255.255.255) and (ip.SrcAddr < 10.0.0.0 or ip.SrcAddr > 10.255.255.255) and (ip.SrcAddr < 192.168.0.0 or ip.SrcAddr > 192.168.255.255) and (ip.SrcAddr < 172.16.0.0 or ip.SrcAddr > 172.31.255.255) and (ip.SrcAddr < 169.254.0.0 or ip.SrcAddr > 169.254.255.255)) or ((ipv6.SrcAddr > ::1) and (ipv6.SrcAddr < 2001::0 or ipv6.SrcAddr >= 2001:1::0) and (ipv6.SrcAddr < fc00::0 or ipv6.SrcAddr >= fe00::0) and (ipv6.SrcAddr < fe80::0 or ipv6.SrcAddr >= fec0::0) and (ipv6.SrcAddr < ff00::0 or ipv6.SrcAddr >= ffff::0)))))
```


В конфиг для openwrt?

**Comment by aldari** [comment]
У меня заработало
--filter-udp=443 в NFQWS_OPT_DESYNC_QUIC
--filter-tcp=80 в NFQWS_OPT_DESYNC_HTTP
--filter-tcp=443 в NFQWS_OPT_DESYNC_HTTPS
если аргумент с путем к файлу пропадает - поместить путь в кавычки.

`nfqws --dpi-desync=split2 --dpi-desync-split-seqovl=1 --dpi-desync-split-tls=sni --wssize 1:6`
C --wssize 1:6 не получится добавить --hostlist="list-general.txt", чтобы влияло только на определенные хосты. У меня подключение начинает отображаться без доступа к интернету.

**Comment by d1agnozzz** [comment]
в соседнем ишшус #455 обсуждалось много конфигов для починки войс-чата, в том числе конфиги для линукса. Испробовал много конфигов от разных людей и через 40 минут страданий наконец заработал войс в дискорде! Ютуб с ним тоже летает. list-general.txt взял из репы Flowseal. Далее идут хайлайты конфига, а не весь конфиг целиком.

```
FWTYPE=nftables

QUIC_PORTS=50000-65535

MODE=nfqws
MODE_HTTP=1
MODE_HTTP_KEEPALIVE=0
MODE_HTTPS=1
MODE_QUIC=1
MODE_FILTER=none

NFQWS_OPT_DESYNC="--dpi-desync=split2 --dpi-desync-ttl=1 --wssize 1:6"

NFQWS_OPT_DESYNC_HTTP="--filter-tcp=80 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new"

NFQWS_OPT_DESYNC_HTTPS="--filter-tcp=443 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake,split --dpi-desync-autottl=2 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new"

NFQWS_OPT_DESYNC_QUIC="--filter-udp=443 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new --filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-cutoff=d3 --dpi-desync-repeats=6 --new"
```

вроде ничего не пропустил, но на всякий случай тут мой конфиг целиком: https://pastebin.com/wxzAStUX

**Comment by boozzze** [comment]
> в соседнем ишшус #455 обсуждалось много конфигов для починки войс-чата, в том числе конфиги для линукса. Испробовал много конфигов от разных людей и через 40 минут страданий наконец заработал войс в дискорде! Ютуб с ним тоже летает. list-general.txt взял из репы Flowseal. Далее идут хайлайты конфига, а не весь конфиг целиком.
> 
> ```
> FWTYPE=nftables
> 
> QUIC_PORTS=50000-65535
> 
> MODE=nfqws
> MODE_HTTP=1
> MODE_HTTP_KEEPALIVE=0
> MODE_HTTPS=1
> MODE_QUIC=1
> MODE_FILTER=none
> 
> NFQWS_OPT_DESYNC="--dpi-desync=split2 --dpi-desync-ttl=1 --wssize 1:6"
> 
> NFQWS_OPT_DESYNC_HTTP="--filter-tcp=80 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new"
> 
> NFQWS_OPT_DESYNC_HTTPS="--filter-tcp=443 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake,split --dpi-desync-autottl=2 --dpi-desync-repeats=6 --dpi-desync-fooling=badseq --dpi-desync-fake-tls=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new"
> 
> NFQWS_OPT_DESYNC_QUIC="--filter-udp=443 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new --filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-cutoff=d3 --dpi-desync-repeats=6 --new"
> ```
> 
> вроде ничего не пропустил, но на всякий случай тут мой конфиг целиком: https://pastebin.com/wxzAStUX

Бро ты единственный кто завёл! Спасибо огромное!
 Тут ниче лишнего? Тока диск и ютуб ? никаких листов?

**Comment by bol-van** [comment]
Собираюсь убирать все это хозяйство с HTTP,HTTPS,QUIC,MODE
Оно создавалось под вариант, когда не было мультистратегий и фильтров.
Сейчас натягивают сову на глобус, пытаясь запихнуть в QUIC незапихуемое только потому, что там udp. Это полнейший misuse и трэш.
Будет несовместимое изменение, полностью ломающее старый конфиг.
Конфигурация будет приближена к winws.
Отдельно настраиваем перехват по портам и отдельно опции единственного инстанса nfqws и/или tpws.
Стандартный autohostlist будет в виде переменной $AUTOHOSTLIST, стандартный hostlist - $HOSTLIST. Куда надо, туда и воткнете. Будете сами думать куда. Как в винде.
Скачивание сетов/листов будет отдельно настраиваться.
Кастомы останутся, но уже без MODE=custom. Достаточно их наличия.
Могут быть варианты, не вписывающиеся даже в новую схему.
Тот же дискорд требует особого ipset. Проводить в конфиг - это слишком частно, такого не будет.
На винде ipset идет через параметр --ipset winws, а на не винде это порочная практика. Естественно, по незнанию и по упрощению будут насиловать несвойственным образом, но это ненормально. Надо использовать kernel ipsets.
Думаю воткну в инсталятор проверку на наличие параметра --ipset и принудительный отказ продолжать.
Иначе один кто-то напишет, а потом по сети этот копипастный трэш будет распространяться.

**Comment by userosos** [comment]
> Думаю воткну в инсталятор проверку на наличие параметра --ipset и принудительный отказ продолжать.

Или вариант лучший - отправить в обработчик непотребство, что бы обработчик сам все сделал как надо, если есть такая возможность конечно.

**Comment by bol-van** [comment]
Не вариант. Использование linux sets сильно отличается от простого параметра.
А эвристить что написал юзер - дело крайне неблагодарное.
Придется ему вместо копипасты напрячься  и скопировать отдельный кастом
А создателю копипасты его написать

**Comment by maityyy** [comment]
@d1agnozzz обновишь конфиг плиз? а то его перелопатили))

**Comment by zefkerr** [comment]
> @d1agnozzz обновишь конфиг плиз? а то его перелопатили))

Запустите ./blockcheck, который определит, какой конфиг для вас лучше всего работает, а не ждите от других копипасты, которая будет работать только у них, но не будет работать у вас.

**Comment by johnnyd-78** [comment]
> NFQWS_OPT_DESYNC_QUIC="--filter-udp=443 --hostlist=/opt/zapret/ipset/list-general.txt --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --new --filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-cutoff=d3 --dpi-desync-repeats=6 --new

Боже, какой устаревший дикий ужас. --filter-udp=50000-65535 > --filter-udp=50000-50090
--dpi-desync=fake,tamper  > --dpi-desync=fake
--dpi-desync-cutoff=d3 > --dpi-desync-cutoff=n2

**Comment by maityyy** [comment]
> > @d1agnozzz обновишь конфиг плиз? а то его перелопатили))
> 
> Запустите ./blockcheck, который определит, какой конфиг для вас лучше всего работает, а не ждите от других копипасты, которая будет работать только у них, но не будет работать у вас.

@Kerrung даже в доках сказано что это не панацея, я не знаю с чего вы решили что конфиг @d1agnozzz не работает у меня, он работал (пока формат не обновили)

**Comment by greno4ka** [comment]
Товарищи, ситуация такова. Одновременно пользуюсь и виндой и линуксом. Под винду форк от flowseal открывает вообще любые сайты на стандартных настройках (general.bat). Под линуксом абсолютно та же конфиругация открывает только дискорд и ютуб. Хочу добиться полной идентичности, а именно "открытие всех сайтов". Видимо топик как раз об этом.
